docs: replace png diagram with mermaid for architecture visualization

scotho3 opened the pull request.

scotho3 committed:
replace png diagram with mermaid diagram for architecture visualization

henderiw commented:
LGTM

ubaumann commented:
Nice work @scotho3 
I think for the mkdocs you need to add the `superfences` extension. I hope it supports `block-beta`.
https://squidfunk.github.io/mkdocs-material/reference/diagrams/

```
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

scotho3 commented:
Thanks @ubaumann - it rendered in native github just fine, I didn't even bother to check it in mkdocs. I'll check that now.

scotho3 committed:
Update mkdocs.yml with markdown and JS enhancements

Added markdown extensions and extra JavaScript for Mermaid diagrams.

scotho3 committed:
Revise Framework.md mermaid diagram and descriptions

Updated the mermaid diagram to reflect changes in the functional building blocks and their relationships.

scotho3 committed:
Delete docs/Framework/images directory

scotho3 commented:
@ubaumann - this has been updated

ubaumann commented:
LGTM

ChrisGrundemann committed:
Merge branch 'main' into patch-1
